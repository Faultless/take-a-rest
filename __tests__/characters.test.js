const axios = require('../__mocks__/axios');

describe('CRUD characters operations', () => {
  it('should get all characters', () =>
    axios.get('example/api/characters').then(res => expect(res.characters).toMatchSnapshot()));
  it('should fail to get characters', () =>
    axios.get('example/api/charactors').catch(err => expect(err).toMatchSnapshot()));
});
