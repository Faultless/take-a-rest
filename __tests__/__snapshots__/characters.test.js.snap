// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUD characters operations should fail to get characters 1`] = `
Object {
  "error": "Bad Request.",
}
`;

exports[`CRUD characters operations should get all characters 1`] = `
Array [
  Object {
    "description": "",
    "id": 1011334,
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg?apikey=498749143bba8e7a297a3589548ed657",
    "name": "3-D Man",
    "rating": 4,
  },
  Object {
    "description": "Rick Jones has been Hulk's best bud since day one, but now he's more than a friend...he's a teammate! Transformed by a Gamma energy explosion, A-Bomb's thick, armored skin is just as strong and powerful as it is blue. And when he curls into action, he uses it like a giant bowling ball of destruction! ",
    "id": 1017100,
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg?apikey=498749143bba8e7a297a3589548ed657",
    "name": "A-Bomb (HAS)",
    "rating": 4,
  },
  Object {
    "description": "AIM is a terrorist organization bent on destroying the world.",
    "id": 1009144,
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/20/52602f21f29ec.jpg?apikey=498749143bba8e7a297a3589548ed657",
    "name": "A.I.M.",
    "rating": 3,
  },
  Object {
    "description": "",
    "id": 1010699,
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg?apikey=498749143bba8e7a297a3589548ed657",
    "name": "Aaron Stack",
    "rating": 0,
  },
]
`;
